framework/doc/content/source/kernels/ConservativeAdvection.md

```diff
@@ -20,29 +20,154 @@ R_i(u_h) = \underbrace{-(\nabla \psi_i, \vec{v} u)}_{\textrm{ConservativeAdvecti
+appropriately.
+
+!alert note
+Full upwinding only works for continuous FEM
+
+In DE systems describing more than just advection (eg, in
+diffusion-advection problems) full upwinding may be used on the
+advection alone.  In practise, it is reasonably common that these more
+complicated cases benefit from numerical stabilization on their
+other terms too, in which case full upwinding could also be used, but
+this is not mandatory and is not implemented in the
+`ConservativeAdvection` Kernel.
+
+For each element in the mesh, full upwinding is implemented in the
+following way.  For each $i$, the quantity
+\begin{equation}
+\tilde{R}_{i} = -(\nabla\psi_i, \vec{v}) \ ,
+\end{equation}
+is evaluated.  If $\tilde{R}_{i}>0$ is positive then mass (or heat, or whatever $u$
+represents) is flowing *out* of node $i$, and this is called an
+"upwind" node.  If $\tilde{R}_{i}\geq 0$, the residual for node $i$
+is set to
+\begin{equation}
+R_{i} = u_{i}\tilde{R}_{i} \ \ \texttt{ for } \ \ \tilde{R}_{i}\geq 0.
+\end{equation}
+Here $u_{i}$ is the value of $u$ at the node $i$.  Define the total
+mass flowing from the upwind nodes:
+\begin{equation}
+M_{\mathrm{out}} = \sum_{i\ \mathrm{ with }\ \tilde{R}_{i}\geq 0}
+u_{i}\tilde{R}_{i} \ .
+\end{equation}
+Similarly, define
+\begin{equation}
+\tilde{M}_{\mathrm{in}} = - \sum_{i \ \mathrm{ with }\  \tilde{R}_{i}< 0}
+\tilde{R}_{i} \ .
+\end{equation}
+Mass is conserved if the residual for the downwind nodes is defined to
+be
+\begin{equation}
+R_{i} = \tilde{R}_{i} M_{\mathrm{out}} / M_{\mathrm{in}}  \ \ \texttt{ for } \ \ \tilde{R}_{i}< 0.
+\end{equation}
+
+Full upwinding adds more numerical diffusion than most other numerical
+stabilization techniques such as SUPG and TVD.  Another problem is
+that steady-state can be hard to achieve in nonlinear problems where
+the velocity is not fixed and changes every nonlinear iteration (this
+does not occur for `ConservativeAdvection`).  On the other hand, full
+upwinding is computationally cheap.
+
+
+
 ## Example Syntax
 
 `ConservativeAdvection` can be used in a variety of problems, including
 advection-diffusion-reaction. The syntax for `ConservativeAdvection` is
-demonstrated in this `Kernels` block from an advection-diffusion-reaction test
-case:
+demonstrated in this `Kernels` block from a pure advection test case:
 
-!listing
-test/tests/dgkernels/advection_diffusion_mixed_bcs_test_resid_jac/dg_advection_diffusion_test.i
-block=Kernels label=false
+!listing test/tests/kernels/conservative_advection/no_upwinding_1D.i block=Kernels
 
-The velocity is supplied as a three component vector with order $v_x$, $v_y$, and  $v_z$.
+The velocity is supplied as a three component vector with order $v_x$,
+$v_y$, and  $v_z$.
+
+## Boundary conditions for pure advection
+
+To form the correct equations, the boundary term $\langle\psi_i, \vec{v} u
+\cdot \vec{n}\rangle$ needs to be included in the `BCs` block of a
+MOOSE input file (n is the outward normal to the boundary).  An
```

Comment:
`s/n/$n$/`, similarly like you do it for $v_x$ above

Reply:
Done